env0/utils.go

```diff
@@ -54,13 +59,21 @@ func readResourceData(i interface{}, d *schema.ResourceData) error {
 		}
 
 		field := val.Field(i)
+
 		fieldType := field.Type()
 
 		dval, ok := d.GetOk(fieldNameSC)
 		if !ok {
 			continue
 		}
 
+		if customField, ok := field.Interface().(CustomResourceDataField); ok {
```

Comment:
a little complex. But basically, if the "type" implements CustomResourceDataField use its functions instead of trying to guess what to do.